### PR TITLE
v1.12 Backports 2023-11-07

### PR DESCRIPTION
**Pull request by jibi**
* [x] #28808 (@brb)
 * [x] #28884 (@julianwiedmann)
 * [x] #28941 (@pchaigno)
 * [x] #28927 (@sayboras)
 * [x] #28642 (@pchaigno)
 * [x] #28795 (@pchaigno)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 28808 28884 28941 28927 28642 28795
```

### CODE REVIEW

**Comment by jibi** [comment]
/test-backport-1.12

**Comment by jibi** [comment]
/test-backport-1.12

**Review by julianwiedmann — APPROVED** [review]

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -481,42 +478,6 @@ func (n *linuxNodeHandler) NodeUpdate(oldNode, newNode nodeTypes.Node) error {
...
-		n.replaceNodeIPSecOutRoute(cidr)
-		if n.nodeConfig.EncryptNode {
-			n.replaceNodeExternalIPSecOutRoute(cidr)
-		}
-	}
-}
-
 // encryptNode handles setting the IPsec state for node encryption (subnet
 // encryption = disabled).
 func (n *linuxNodeHandler) encryptNode(newNode *nodeTypes.Node) {
```
Comment (pchaigno) [inline pkg/datapath/linux/node.go]: We should also move this function. I'll fix it.

**Comment by pchaigno** [comment]
/test-backport-1.12

**Comment by pchaigno** [comment]
/test-backport-1.12

**Review by sayboras — APPROVED** [review]
Looks good for my commit, thanks ✅

**Comment by jibi** [comment]
/test-backport-1.12

**Review by brb — APPROVED** [review]
Thanks!

**Comment by jibi** [comment]
/test-1.16-4.9

**Comment by jibi** [comment]
/test-1.24-net-next

**Review by pchaigno — APPROVED** [review]
My PR you backported looks good. Thanks!

**Pull request merged**